examples_utils/benchmarks/run_benchmarks.py

```diff
@@ -108,13 +122,32 @@ def proc_thread():
 
     t = threading.Thread(target=proc_thread, name="proc_thread")
     t.start()
+    if monitor_ipus:
```

Comment:
Not an issue, but do we need to conditionally define a function? I cant see a risk of it being called without being in an `if monitor_ipus`

More of a design decision thing. No complaints or anything

Reply:
It looks a bit ugly, I was trying to make REALLY sure I didn't change the behaviour when `--gc-monitor` was not specified.

I think the positioning of the function was based on a bit of misconception of mine, I thought I _needed_ `t` to be defined before `monitor_thread`, but that's not quite right it can be defined before, as long as it's not called before `t` is defined. It's a bit messy but I'm not sure how to improve the design.

Reply:
Since I'm adding a comment I'll take it out of the if statement.